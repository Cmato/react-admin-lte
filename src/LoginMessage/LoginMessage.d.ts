import * as React from 'react';

export interface LoginMessageProps {
  component?: React.ReactType<LoginMessageProps>;
}

declare const LoginMessage: React.ComponentType<LoginMessageProps>;

export default LoginMessage;
