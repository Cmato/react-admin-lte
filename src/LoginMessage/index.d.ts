export { default } from './LoginMessage';
export * from './LoginMessage';
