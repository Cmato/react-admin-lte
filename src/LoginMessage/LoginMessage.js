import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const LoginMessage = ({
  children,
  className: classNameProp,
  component: Component,
  ...other
}) => {
  const className = classNames({
    'login-box-msg': true
  }, classNameProp);

  return (
    <Component className={className} {...other}>
      {children}
    </Component>
  )
};

LoginMessage.propTypes = {
  /**
   * The contents of the component.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
};

LoginMessage.defaultProps = {
  component: 'p'
};

export default LoginMessage;
