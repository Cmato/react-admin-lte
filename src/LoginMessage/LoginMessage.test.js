import React from 'react';
import { shallow } from 'enzyme';
import LoginMessage from './LoginMessage';

it('renders default', () => {
  expect(shallow(<LoginMessage>Sign in</LoginMessage>))
    .toMatchSnapshot();
});

it('renders custom component', () => {
  expect(shallow(<LoginMessage component="div">Sign in</LoginMessage>))
    .toMatchSnapshot();
});
