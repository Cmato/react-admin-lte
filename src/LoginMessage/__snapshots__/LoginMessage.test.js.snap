// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom component 1`] = `
<div
  className="login-box-msg"
>
  Sign in
</div>
`;

exports[`renders default 1`] = `
<p
  className="login-box-msg"
>
  Sign in
</p>
`;
